pkg/kubelet/cm/node_container_manager_linux_test.go

```diff
@@ -400,3 +401,47 @@ func getEphemeralStorageResourceList(storage string) v1.ResourceList {
 	}
 	return res
 }
+
+func TestUpdateNodeAllocatableCgroups(t *testing.T) {
+	tests := []struct {
+		name             string
+		internalCapacity v1.ResourceList
+		updatedList      v1.ResourceList
+		expectedChange   bool
+	}{
+		{
+			name:             "Empty capacity",
+			internalCapacity: v1.ResourceList{},
+			updatedList:      v1.ResourceList{},
+		},
+		{
+			name:             "irrelevant capacity update",
+			internalCapacity: v1.ResourceList{},
+			updatedList:      getResourceList("10", "10Gi"),
+		},
+		{
+			name:             "CPU updated",
+			internalCapacity: getResourceList("12", "10Gi"),
+			updatedList:      getResourceList("10", "10Gi"),
+			expectedChange:   true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			node := &v1.Node{}
+			node.Name = "test"
+			cm := &containerManagerImpl{
+				internalCapacity: test.internalCapacity.DeepCopy(),
+				nodeInfo:         node,
+			}
+
+			_ = cm.UpdateNodeAllocatableCgroups(test.updatedList)
```

Comment:
What is going to happen during testing mode when we want to `enforceNodeAllocatableCgroups`?

Should we be checking the error return value here?

Reply:
`UpdateNodeAllocatableCgroups` doesn't generates errors, but simply propagates the ones happened in `enforceNodeAllocatableCgroups`. Checking the error value implies testing `enforceNodeAllocatableCgroups` which, I think, should be covered with its own unit tests. Currently it's not. Testing it would require mocking `runc` used underneath.

`enforceNodeAllocatableCgroups` is not the culprit of the fixed bug, so I decided to leave it out of this PR's scope. How about adding a TODO comment here for `enforceNodeAllocatableCgroups` tests?

Of course I can add the tests, but it'll take time.

Reply:
Whoops I had a typo in my question - I meant to ask, what is going to happen when the `enforceNodeAllocatableCgroups` code executes in the unit test environment?

I'm comfortable not adding testing for it now as it wasn't the source of the bug - also +1 on adding a comment/separate pr indicating it should be tested in the future.

Reply:
Since `cm` doesn't have `cgroupRoot` set up `enforceNodeAllocatableCgroups` does nothing and simply returns `nil`.

I've added the TODO comment. Thank you!